sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorLiveTestBase.cs

```diff
@@ -88,5 +89,14 @@ protected void ValidateGroupKey(DimensionKey groupKey)
                 Assert.That(column.Value, Is.Not.Null.And.Not.Empty);
             }
         }
+
+        private MetricsAdvisorClientsOptions GetInstrumentedOptions()
+        {
+            var options = new MetricsAdvisorClientsOptions();
+
+            options.Retry.MaxRetries = 6;
```

Comment:
Doubling the default maximum number of retries allowed by our retry policy (default = 3).

At first, my original approach was trying to use an `HttpPipelinePolicy` to keep retrying as long as we got 429 errors back from the service (you can check it [here](https://github.com/Azure/azure-sdk-for-net/pull/20653/commits/76844850801db52705ef1ad5e207f340bd0638d4)). It worked just fine.

However, I have changed the approach to something simpler (and a bit less deterministic). The `HttpPipelinePolicy` could end up in an endless loop if the service kept returning a 429 for any reason (for example, a bug on their side).

Increasing the maximum number of retries seems to be enough to solve the problem. [Here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/ResponseClassifier.cs#L17) is the list of status codes affected by this change. These are not critical errors and will be retried 3 more times in case of failure.

I have run the live tests pipeline 3 times with the `MaxRetries` approach and had zero failures, which is great given that we were hitting the problem in 100% of the runs. We can easily increase the number if we start seeing the error again, or consider using the `HttpPipelinePolicy` again if the error persists.

Reply:
Out of curiosity, are the other languages facing this too? And are the solutions the same? I ask because this seems generic enough that according to how their pipelines are set, it will apply to everyone else.

Reply:
I raised this concern during standup some weeks ago when the issue started, but no other languages seemed to be facing it at the time. It depends on how the pipelines are set like you said, and on the number of tests as well.